### PR TITLE
[SYCL] Fix check-sycl-deploy target problems

### PR DESCRIPTION
**Pull request by Fznamznon**
- deploy-lit.site.cfg configuration file used to owerwrite llvm-lit to
      use binaries from deploy, which sometimes do not exist. Removed
      deploy-lit.site.cfg and made all configuration files flexible using
      llvm-lit command line options.
    - check-all target used to run check-sycl-deploy target. This caused
      problems if CMAKE_INSTALL_PREFIX wasn't specified, because in this
      case check-sycl-deploy target leads to trying to install sycl binaries
      to system directories. Excluded check-sycl-deploy target from
      check-all.
    - Reduced copy-paste in check-sycl-deploy target dependencies.
    - Reduced number of variables passed from cmake to lit.

Signed-off-by: Mariya Podchishchaeva <mariya.podchishchaeva@intel.com>

### CODE REVIEW

File: sycl/test/CMakeLists.txt [hunk 1]
```diff
@@ -37,7 +37,7 @@ configure_lit_site_cfg(
   ${CMAKE_CURRENT_SOURCE_DIR}/deploy-lit.site.cfg.py.in
   ${CMAKE_CURRENT_BINARY_DIR}/deploy-lit.site.cfg.py
   MAIN_CONFIG
-  ${CMAKE_CURRENT_SOURCE_DIR}/lit.cfg.py
+  ${CMAKE_CURRENT_SOURCE_DIR}/deploy-lit.cfg.py
```
Comment (AlexeySachkov) [inline sycl/test/CMakeLists.txt]: I really don't like the idea that we had to copy this file and sync it in order to get all fixes/improvements.

I suggest to perform the same trick which were used to launch LIT tests in-tree back in 2017: https://github.com/llvm/llvm-project/commit/ce92db13ead9a24c767d60375545fa6ad6cb8ce0#diff-707931295fba376ca52d0861200e7adaL25.

In particular:
- re-order `configure_lit_site_cfg` directives, so, by default mapping is set for regular LIT tests
- add `--override-site-cfg` option or something like this which will allow to override default mapping `lit.cfg` -> `lit.site.cfg` for `check-sycl-deploy`

File: sycl/test/devicelib/c99_complex_math_fp64_test.cpp [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // UNSUPPORTED: windows
 // RUN: %clangxx -fsycl -c %s -o %t.o
-// RUN: %clangxx -fsycl %t.o %llvm_build_libs_dir/libsycl-complex-fp64.o -o %t.out
+// RUN: %clangxx -fsycl %t.o %sycl_libs_dir/libsycl-complex-fp64.o -o %t.out
```
Comment (bader) [inline sycl/test/devicelib/c99_complex_math_fp64_test.cpp]: Why do we need to use `%sycl_libs_dir/` prefix?
I thought that clang driver is aware about location of deployed libraries. Am I wrong? 
Do we use non-standard directories?
Why?

File: sycl/test/CMakeLists.txt (hunk 1 above)
Comment (Fznamznon) [reply sycl/test/CMakeLists.txt]: I used it to remove deploy-lit.cfg.py at all. Take a look.

File: sycl/test/devicelib/c99_complex_math_fp64_test.cpp (hunk 2 above)
Comment (Fznamznon) [reply sycl/test/devicelib/c99_complex_math_fp64_test.cpp]: I'm not an expert in device libraries and clang driver. But for me it looks like that technically `libsycl-complex-fp64.o` is not a library, but some object file. The clang driver seems ok to find deployed `libsycl.so` library, but not these object files.
Tagging @asavonic , @mdtoguchi , @AGindinson who may know a bit more than me.

**Comment by Fznamznon** [comment]
The clang-format failure doesn't seem relevant to fix in my patch.

**Comment by alexbatashev** [comment]
> The clang-format failure doesn't seem relevant to fix in my patch.

@Fznamznon you need to merge sycl into your branch to get fixes for this job.

**Comment by bader** [comment]
@Fznamznon, please, ignore https://github.com/intel/llvm/pull/1165/checks?check_run_id=472335088 results. I'll exclude test directory form this check.

File: sycl/test/lit.site.cfg.py.in [hunk 3]
```diff
@@ -2,18 +2,16 @@
...
-config.llvm_build_bins_dir = "@LLVM_BUILD_BINARY_DIRS@"
-config.llvm_binary_dir = "@LLVM_BINARY_DIR@"
-config.sycl_include = "@SYCL_INCLUDE@"
+config.sycl_include = lit_config.params.get('SYCL_INCLUDE', "@SYCL_INCLUDE@")
 config.sycl_obj_root = "@SYCL_BINARY_DIR@"
 config.sycl_source_dir = "@SYCL_SOURCE_DIR@/source"
 config.opencl_libs_dir = os.path.dirname("@OpenCL_LIBRARIES@")
+config.sycl_libs_dir = lit_config.params.get('SYCL_LIBS_DIR', "@LLVM_LIBS_DIR@")
+config.target_triple = "@TARGET_TRIPLE@"
+config.host_triple = "@LLVM_HOST_TRIPLE@"
```
Comment (AlexeySachkov) [inline sycl/test/lit.site.cfg.py.in]: I cannot find direct uses of these new options, are they needed?

Comment (Fznamznon) [reply sycl/test/lit.site.cfg.py.in]: `llvm_config.use_clang()` function call requires them to be set.

File: sycl/test/devicelib/c99_complex_math_fp64_test.cpp (hunk 2 above)
Comment (mdtoguchi) [reply sycl/test/devicelib/c99_complex_math_fp64_test.cpp]: as `libsycl.so` is considered a library, the linker will go ahead and search for it via the either the default `-L<dir>` values or `-L<dir>` values passed on the command line.  Objects need to be fully qualified or in the `$CWD`

File: sycl/test/lit.site.cfg.py.in (hunk 3 above)
Comment (bader) [reply sycl/test/lit.site.cfg.py.in]: @AlexeySachkov, please, "resolve" conversation.

File: sycl/test/devicelib/c99_complex_math_fp64_test.cpp (hunk 2 above)
Comment (bader) [reply sycl/test/devicelib/c99_complex_math_fp64_test.cpp]: I expect this functionality to be implemented as a library and handled by the driver.
Let's discuss it separately.

**Comment by bader** [comment]
> @Fznamznon, please, ignore https://github.com/intel/llvm/pull/1165/checks?check_run_id=472335088 results. I'll exclude test directory form this check.

Fixed by #1224.

File: sycl/test/CMakeLists.txt [hunk 4]
```diff
@@ -1,23 +1,11 @@
...
-set(CLANGCL_IN_DEPLOY "${CMAKE_INSTALL_PREFIX}/bin/clang-cl")
 
 get_target_property(SYCL_BINARY_DIR sycl-toolchain BINARY_DIR)
 
 set(SYCL_INCLUDE "${dst_dir}")
-set(SYCL_DEPLOY_INCLUDE "${dst_deploy_dir}")
 
 set(RT_TEST_ARGS ${RT_TEST_ARGS} "-v")
-set(DEPLOY_RT_TEST_ARGS ${DEPLOY_RT_TEST_ARGS} "-v --config-prefix=deploy-lit")
+set(DEPLOY_RT_TEST_ARGS ${DEPLOY_RT_TEST_ARGS} "-v -D SYCL_TOOLS_DIR=${CMAKE_INSTALL_PREFIX}/bin -D SYCL_LIBS_DIR=${CMAKE_INSTALL_PREFIX}/lib -D SYCL_INCLUDE=${dst_deploy_dir}")
```
Comment (fwyzard) [inline sycl/test/CMakeLists.txt]:
```suggestion
set(DEPLOY_RT_TEST_ARGS ${DEPLOY_RT_TEST_ARGS} "-v -D SYCL_TOOLS_DIR=${CMAKE_INSTALL_PREFIX}/bin -D SYCL_LIBS_DIR=${CMAKE_INSTALL_PREFIX}/lib${LLVM_LIBDIR_SUFFIX} -D SYCL_INCLUDE=${dst_deploy_dir}")
```

Comment (bader) [reply sycl/test/CMakeLists.txt]: I'm going to commit it separately. Thanks!

Comment (bader) [reply sycl/test/CMakeLists.txt]: https://github.com/intel/llvm/pull/1228

**Pull request merged**